main.cc

```diff
@@ -11,22 +12,33 @@ void usage(char* prog) {
   std::cout << "    -h         : help" << std::endl;
   std::cout << "    -l         : list all keys and values" << std::endl;
   std::cout << "    -v         : version" << std::endl;
+  std::cout << "    -n         : tries to query the temperature sensor for n times";
+  std::cout << " (1 second interval) until a valid value is returned" << std::endl;
 }
 
 int main(int argc, char *argv[]) {
   int c;
-  extern char   *optarg;
+  unsigned int attempts = 1;
 
   kern_return_t result;
   int           op = smctemp::kOpNone;
   smctemp::UInt32Char_t  key = { 0 };
   smctemp::SmcVal_t      val;
 
-  while ((c = getopt(argc, argv, "clvh")) != -1) {
+  while ((c = getopt(argc, argv, "clvhn::")) != -1) {
```

Comment:
It seems that double colons are a GNU extension, and on one of my two Macs unfortunately, this doesn't work as expected. 
```console
$ ./smctemp -n1 -c
65.8

$ ./smctemp -n 1 -c
Check Temperature by using Apple System Management Control (Smc) tool 0.1.1
Usage:
./smctemp [options]
    -c         : list CPU temperatures (Celsius)
    -h         : help
    -l         : list all keys and values
    -v         : version
    -n         : tries to query the temperature sensor for n times (1 second interval) until a valid value is returned
```

Ref. https://man7.org/linux/man-pages/man3/getopt.3.html#DESCRIPTION
> Two colons
>       mean an option takes an optional arg; if there is text in the
>       current argv-element (i.e., in the same word as the option name
>       itself, for example, "-oarg"), then it is returned in optarg,
>       otherwise optarg is set to zero.  This is a GNU extension. 

How do you think about using a single colon and making the argument for -n mandatory?
```suggestion
  while ((c = getopt(argc, argv, "clvhn:")) != -1) {
```

Reply:
I think it is fine. The only drawback is that it somehow breaks the "contract" for those already using simply `./smctemp -c`.
But I personally think it's fine since the project is still quite new. I'll adjust it in a few hours.